scripts/setup-amazon-linux-2.sh

```diff
@@ -0,0 +1,337 @@
+function sudo_cmake_install {
+  local NAME=$(basename "$(pwd)")
+  local BINARY_DIR=_build
+
+  if [ -d "${BINARY_DIR}" ] && prompt "Do you want to rebuild ${NAME}?"; then
+    rm -rf "${BINARY_DIR}"
+  fi
+  mkdir -p "${BINARY_DIR}"
+  CPU_TARGET="${CPU_TARGET:-avx}"
+  COMPILER_FLAGS=$(get_cxx_flags $CPU_TARGET)
+
+  # CMAKE_POSITION_INDEPENDENT_CODE is required so that Velox can be built into dynamic libraries \
+  cmake -Wno-dev -B"${BINARY_DIR}" \
+    -GNinja \
+    -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
+    -DCMAKE_CXX_STANDARD=17 \
+    "${INSTALL_PREFIX+-DCMAKE_PREFIX_PATH=}${INSTALL_PREFIX-}" \
+    "${INSTALL_PREFIX+-DCMAKE_INSTALL_PREFIX=}${INSTALL_PREFIX-}" \
+    -DCMAKE_CXX_FLAGS="$COMPILER_FLAGS" \
+    -DBUILD_TESTING=OFF \
+    "$@"
+
+  # install a prebuilt project with elevated privileges
+  # This is useful for systems where /usr/ is not writable
+  local BINARY_DIR=_build
+  sudo ninja-build -C "${BINARY_DIR}" install
+}
+
+function clean_dir {
+  local DIRNAME=$(basename $1)
+  if [ -d "${DIRNAME}" ]; then
+    rm -rf "${DIRNAME}"
+  fi
+  mkdir ${DIRNAME}
+}
+
+function run_and_time {
+  time "$@"
+  { echo "+ Finished running $*"; } 2> /dev/null
+}
+
+function git_clone {
+  local NAME=$1
+  shift
+  local REPO=$1
+  shift
+  local GIT_CLONE_PARAMS=$@
+  local DIRNAME=$(basename $NAME)
+  if [ -d "${DIRNAME}" ]; then
+    rm -rf "${DIRNAME}"
+  fi
+  git clone -q $GIT_CLONE_PARAMS "${REPO}"
+}
+
+function prompt {
+  (
+    while true; do
+      local input="${PROMPT_ALWAYS_RESPOND:-}"
+      echo -n "$(tput bold)$* [Y, n]$(tput sgr0) "
+      [[ -z "${input}" ]] && read input
+      if [[ "${input}" == "Y" || "${input}" == "y" || "${input}" == "" ]]; then
+        return 0
+      elif [[ "${input}" == "N" || "${input}" == "n" ]]; then
+        return 1
+      fi
+    done
+  ) 2> /dev/null
+}
+
+function install_checkinstall {
+  if [[ -v SKIP_CHECKINSTALL ]]
+  then
+      return 0
+  fi
+
+  git_clone checkinstall http://checkinstall.izto.org/checkinstall.git
+  cd checkinstall
+  make
+  sudo make install
+  cd ..
+}
+
+function install_cmake {
+  if [[ -v SKIP_CMAKE ]]
+  then
+      return 0
+  fi
+
+  local DIRNAME=$(basename "cmake-3.25.1")
+  clean_dir ${DIRNAME}
+  pushd ./
+  cd ${DIRNAME}
+  wget https://github.com/Kitware/CMake/releases/download/v3.25.1/cmake-3.25.1-linux-x86_64.sh
+  chmod +x cmake-3.25.1-linux-x86_64.sh
+  sudo ./cmake-3.25.1-linux-x86_64.sh --prefix=/usr/local/ --skip-license
+  popd
+}
+
+function install_ccache {
+  if [[ -v SKIP_CCACHE ]]
+  then
+      return 0
+  fi
+  local DIRNAME=$(basename "ccache-for-velox")
+  if [ -d "${DIRNAME}" ];
+  then
+    rm -rf "${DIRNAME}"
+  fi
+
+  pushd ./
+
+  mkdir -p ${DIRNAME}
+  cd ${DIRNAME}
+
+  wget https://github.com/ccache/ccache/releases/download/v4.7.4/ccache-4.7.4-linux-x86_64.tar.xz
+  tar -xf ccache-4.7.4-linux-x86_64.tar.xz
+  sudo cp ccache-4.7.4-linux-x86_64/ccache /usr/local/bin/
+
+  popd
+}
+
+function install_boost {
```

Comment:
FYI, With the latest main, boost need not be installed on the system, and will be built as part of Velox, would be great if this can be removed.

Reply:
Ok, I'll give that a try. Thanks!